src/node_errors.cc

```diff
@@ -616,8 +616,18 @@ v8::ModifyCodeGenerationFromStringsResult ModifyCodeGenerationFromStrings(
     bool is_code_like) {
   HandleScope scope(context->GetIsolate());
 
+  if (context->GetNumberOfEmbedderDataFields() <=
```

Comment:
I think `ContextEmbedderTag::IsNodeContext(context)` may be a more robust check here.

Reply:
The two are not assigned together, one does not imply the other especially if they are assigned by embedders, so I think we should do a quick check for the longer one, without the actual filed for modified code generation I don’t think there is enough information to do anything.

Reply:
(Also, the `env == nullptr` check below already covers the ContextEmbedderTag check, the embedder data field number check is just a faster failure path)